docs/core/logging.md

```diff
@@ -178,148 +224,286 @@ Powertools for AWS Lambda (Java) extends the functionality of Log4J. Below is an
 
-    ```properties hl_lines="1 2"
-    log4j.layout.jsonTemplate.timestampFormatPattern=yyyy-MM-dd'T'HH:mm:ss.SSSZz
-    log4j.layout.jsonTemplate.timeZone=Europe/Oslo
+=== "AppLogResponse.java"
+
+    ```java hl_lines="5"
+    public class AppLogResponse implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    
+        private static final Logger LOGGER = LoggerFactory.getLogger(AppLogResponse.class);
+        
+        @Logging(logResponse = true)
+        public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
+            // ...
+        }
+    }
     ```
 
-- Utility also provides sample template for [Elastic Common Schema(ECS)](https://www.elastic.co/guide/en/ecs/current/ecs-reference.html) layout.
-The field emitted in logs will follow specs from [ECS](https://www.elastic.co/guide/en/ecs/current/ecs-reference.html) together with field captured by utility as mentioned [above](#standard-structured-keys).
+### Logging handler uncaught exception
+By default, AWS Lambda logs any uncaught exception that might happen in the handler. However, this log is not structured
+and does not contain any additional context. You can instruct the `@Logging` annotation to log this kind of exception 
+with `logError` param or via `POWERTOOLS_LOGGER_LOG_ERROR` env var.
 
-    Use `LambdaEcsLayout.json` as `eventTemplateUri` when configuring `JsonTemplateLayout`.
+???+ warning
+    This is disabled by default to prevent double logging
 
-=== "log4j2.xml"
+=== "AppLogResponse.java"
 
-    ```xml hl_lines="5"
-    <?xml version="1.0" encoding="UTF-8"?>
-    <Configuration>
-        <Appenders>
-            <Console name="JsonAppender" target="SYSTEM_OUT">
-                <JsonTemplateLayout eventTemplateUri="classpath:LambdaEcsLayout.json" />
-            </Console>
-        </Appenders>
-        <Loggers>
-            <Logger name="JsonLogger" level="INFO" additivity="false">
-                <AppenderRef ref="JsonAppender"/>
-            </Logger>
-            <Root level="info">
-                <AppenderRef ref="JsonAppender"/>
-            </Root>
-        </Loggers>
-    </Configuration>
+    ```java hl_lines="5"
+    public class AppLogError implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    
+        private static final Logger LOGGER = LoggerFactory.getLogger(AppLogError.class);
+        
+        @Logging(logError = true)
+        public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
+            // ...
+        }
+    }
     ```
 
-## Setting a Correlation ID
+### Logging additional keys
 
-You can set a Correlation ID using `correlationIdPath` attribute by passing a [JSON Pointer expression](https://datatracker.ietf.org/doc/html/draft-ietf-appsawg-json-pointer-03){target="_blank"}.
+#### Logging a correlation ID
```

Comment:
Worth mentioning where you'd expect to find the correlation ID? We had this chat the other day with the "why isn't it in the environment variable anymore?" thread

Reply:
not sure to understand...

Reply:
we reference the known correlation ids later...

Reply:
it's in the same chapter... a few lines below